assets/html/themeswap.js

```diff
@@ -25,8 +31,12 @@ function set_theme_from_local_storage() {
     // To distinguish the default (primary) theme, it needs to have the data-theme-primary
     // attribute set.
     var isprimary = (ss.ownerNode.getAttribute("data-theme-primary") !== null);
+    // Check if the theme is primary dark theme
+    var isDarkTheme = (ss.ownerNode.getAttribute("dark-theme-primary") !== null);
+    // Use ss is for dark theme then set the value of darkTheme
+    if(isDarkTheme) darkTheme = themename;
     // If we find a matching theme (and it's not the default), we'll set active to non-null
-    if(!isprimary && themename === theme) active = i;
+    if(themename === theme) active = i;
```

Comment:
I think keeping the `isprimary` check here makes sense, because we don't want to mess around with applying class names when it's the primary light theme. Unless I am missing something?

```suggestion
    if(!isprimary && themename === theme) active = i;
```

Reply:
The problem here is if I prefer dark color scheme but I want to keep light theme for the site then I basically set the theme manually in the local storage. Since the theme is primary we ignore this `if` condition and then it moves on and basically uses my preference of dark scheme. We somehow need an override over the color preference in case of primary theme.

Reply:
Ah, you're right, never mind.